lib/keypress.js

```diff
@@ -198,22 +198,20 @@ keypress.listen = (options = {}, onKeypress) => {
   if (!stdin || (stdin !== process.stdin && !stdin.isTTY)) {
     throw new Error('Invalid stream passed');
   }
+  emitKeypressEvents(stdin)
 
-  let rl = readline.createInterface({ terminal: true, input: stdin });
-  readline.emitKeypressEvents(stdin, rl);
-
-  let on = (buf, key) => onKeypress(buf, keypress(buf, key), rl);
+  let on = (buf, key) => onKeypress(buf, keypress(buf, key));
   let isRaw = stdin.isRaw;
 
   if (stdin.isTTY) stdin.setRawMode(true);
   stdin.on('keypress', on);
-  rl.resume();
+  stdin.resume();
 
   let off = () => {
     if (stdin.isTTY) stdin.setRawMode(isRaw);
     stdin.removeListener('keypress', on);
-    rl.pause();
-    rl.close();
+    stdin.pause()
```

Comment:
thanks for this fix. I've wanted to get rid of the superfluous `.close()` for a while and figured I'd just fix in the upcoming release. I'm glad this will be fixed in the current major as well.